src/source/PeerConnection/Rtp.c

```diff
@@ -181,19 +181,27 @@ STATUS writeFrame(PRtcRtpTransceiver pRtcRtpTransceiver, PFrame pFrame)
     CHK_STATUS(constructRtpPackets(pPayloadArray, pKvsRtpTransceiver->sender.payloadType, pKvsRtpTransceiver->sender.sequenceNumber, rtpTimestamp, pKvsRtpTransceiver->sender.ssrc, pPacketList, pPayloadArray->payloadSubLenSize));
     pKvsRtpTransceiver->sender.sequenceNumber = GET_UINT16_SEQ_NUM(pKvsRtpTransceiver->sender.sequenceNumber + pPayloadArray->payloadSubLenSize);
 
+    bufferAfterEncrypt = (pKvsRtpTransceiver->sender.payloadType == pKvsRtpTransceiver->sender.rtxPayloadType);
     for (i = 0; i < pPayloadArray->payloadSubLenSize; i++) {
         pRtpPacket = pPacketList + i;
         CHK_STATUS(createBytesFromRtpPacket(pRtpPacket, &rawPacket, &packetLen));
-        rawLen = packetLen;
 
+        if (!bufferAfterEncrypt) {
+            pRtpPacket->pRawPacket = rawPacket;
+            pRtpPacket->rawPacketLength = packetLen;
+            CHK_STATUS(rtpRollingBufferAddRtpPacket(pKvsRtpTransceiver->sender.packetBuffer, pRtpPacket));
+        }
+
+        signedPacketLen = packetLen;
         rawPacket = MEMREALLOC(rawPacket, packetLen + 10); // For SRTP authentication tag
-        pRtpPacket->pRawPacket = rawPacket;
-        pRtpPacket->rawPacketLength = packetLen;
-        if (pKvsRtpTransceiver->sender.packetBuffer != NULL) {
+        CHK_STATUS(encryptRtpPacket(pKvsPeerConnection->pSrtpSession, rawPacket, &signedPacketLen));
```

Comment:
Shouldn't these be in an else clause of !bufferAfterEncrypt? I am just not following the logic perhaps.

Reply:
We always need to encrypt+send. I don't think we will be able to do this in an if/else clause. It is ended to be.

* If we aren't storing the encrypted packet copy the packet and store
* encrypt
* If we are storing the encrypted packet copy the packet and store